File: simulation/generator.js
const { faker } = require('@faker-js/faker');

function generatePatientData(numPatients = 50) {
    const patients = [];

    // Base start time for the day (8:00 AM)
    const baseTime = new Date();
    baseTime.setHours(8, 0, 0, 0);

    for (let i = 0; i < numPatients; i++) {
        // Stagger arrival times
        const arrivalBias = i * (Math.floor(Math.random() * (15 - 5 + 1)) + 5); // 5 to 15 minutes
        const arrivalTime = new Date(baseTime.getTime() + arrivalBias * 60000);

        const appointmentScheduled = new Date(arrivalTime.getTime() - (Math.floor(Math.random() * 14) + 1) * 24 * 60 * 60 * 1000);

        const conditions = ['Routine Checkup', 'Flu Symptoms', 'Hypertension', 'Diabetes Follow-up', 'Acute Pain'];

        const patient = {
            patient_id: faker.string.uuid().substring(0, 8),
            name: faker.person.fullName(),
            age: Math.floor(Math.random() * (90 - 18 + 1)) + 18,
            condition: conditions[Math.floor(Math.random() * conditions.length)],
            arrival_time: arrivalTime,
            appointment_scheduled: appointmentScheduled
        };
        patients.push(patient);
    }

    return patients;
}

module.exports = { generatePatientData };

File: simulation/generator.py
import pandas as pd
import numpy as np
from faker import Faker
from datetime import datetime, timedelta
import random

fake = Faker()

def generate_patient_data(num_patients=50, random_seed=42):
    random.seed(random_seed)
    np.random.seed(random_seed)
    Faker.seed(random_seed)

    patients = []
    
    # Base start time for the day (8:00 AM)
    base_time = datetime.now().replace(hour=8, minute=0, second=0, microsecond=0)

    for i in range(num_patients):
        # Stagger arrival times
        arrival_bias = i * random.randint(5, 15) # minutes between arrivals
        arrival_time = base_time + timedelta(minutes=arrival_bias)
        
        patient = {
            'patient_id': fake.uuid4()[:8],
            'name': fake.name(),
            'age': random.randint(18, 90),
            'condition': random.choice(['Routine Checkup', 'Flu Symptoms', 'Hypertension', 'Diabetes Follow-up', 'Acute Pain']),
            'arrival_time': arrival_time,
            'appointment_scheduled': arrival_time - timedelta(days=random.randint(1, 14))
        }
        patients.append(patient)
    
    return pd.DataFrame(patients)
